### ISSUE TITLE
problems compiling the renderer in openWRT

### ISSUE THREAD

**Issue by VincentKuscher** [hzeller/gmrender-resurrect#107]
There is a problem with the webserver.c file with the new openWRT version.
# include <error.h> in webserver.c is not working anymore.

It seems that the error.h file of the gnu lib is missing.
I deleted the include and uncommented all error() messages in webserver.c

Another issue is that the define of PATH_MAX is missing in webserver.c
I added #define MAX_PATH 200 in webserver.h and the compilation runs fine.

**Comment by tristan-k** [comment]
Can you give some more instructions on how to compile for openwrt.

**Comment by HugoPoi** [comment]
Same here, I try to compile for openwrt (master) and get 
```
webserver.c:34:19: fatal error: error.h: No such file or directory
compilation terminated.
```
What package or dependencie provide the `<error.h>` libc no?

@tristan-k you succecefully build with this kind of setup ? i can provide more info about my build if needed

**Comment by hzeller** [comment]
Could you try again with a fresh `git pull` ?

**Comment by HugoPoi** [comment]
Missing `PATH_MAX` constant now.
```
webserver.c: In function 'webserver_register_file':
webserver.c:89:19: error: 'PATH_MAX' undeclared (first use in this function)
  char local_fname[PATH_MAX];
                   ^
webserver.c:89:19: note: each undeclared identifier is reported only once for each function it appears in
webserver.c:89:7: warning: unused variable 'local_fname' [-Wunused-variable]
  char local_fname[PATH_MAX];
       ^
make[5]: *** [Makefile:391: webserver.o] Error 1
```

Adding `#define PATH_MAX 200` in webserver.h resolve this.
I don't if it's a good value ?

**Comment by hzeller** [comment]
200 should be fine.
I now got rid of of the use of PATH_MAX, so if you do a `git pull` again, it should work.
